alchemist-full/build.gradle.kts

```diff
@@ -30,6 +37,99 @@ application {
     mainClass.set("it.unibo.alchemist.Alchemist")
 }
 
+// JPackage
+tasks.jpackage {
+    // General info
+    resourceDir = "${rootProject.projectDir}/package-settings"
+    appName = rootProject.name
+    appVersion = rootProject.version.toString().substringBefore('-')
+    copyright = "Copyright (C) 2010-2023, Danilo Pianini and contributors"
+    appDescription = rootProject.description
+    vendor = ""
+    licenseFile = "${rootProject.projectDir}/LICENSE.md"
+    verbose = isInCI
+
+    // Packaging settings
+    input = rootProject.layout.buildDirectory.map { it.dir("package-input") }.get().asFile.path
+    destination = rootProject.layout.buildDirectory.map { it.dir("package") }.get().asFile.path
+    mainJar = "alchemist-full-${rootProject.version}-all.jar"
+    mainClass = "it.unibo.alchemist.Alchemist"
+
+    linux {
+        icon = "${rootProject.projectDir}/package-settings/logo.png"
+        type = ImageType.RPM
+    }
+    windows {
+        icon = "${rootProject.projectDir}/package-settings/logo.ico"
+        type = ImageType.MSI
+        winDirChooser = true
+        winShortcutPrompt = true
+        winPerUserInstall = isInCI
+    }
+    mac {
+        icon = "${rootProject.projectDir}/package-settings/logo.png"
+        type = ImageType.PKG
+    }
+
+    dependsOn("copyForPackaging")
+}
+
+tasks.register<Copy>("copyForPackaging") {
+    val jarFile = "alchemist-full-${rootProject.version}-all.jar"
+    from("${rootProject.projectDir}/build/shadow/$jarFile")
+    into("${rootProject.projectDir}/build/package-input")
+    dependsOn("shadowJar")
+}
+
+tasks.register<Exec>("testJpackageOutput") {
+    group = "Verification"
+    description = "Verifies the jpackage output correctness for the OS running the script"
+    val interceptOutput = ByteArrayOutputStream()
+    val interceptError = ByteArrayOutputStream()
```

Comment:
why do you intercept output and error?

Reply:
Dead code from old implementations, i'm going to remove it